### PR TITLE
CI optimizations

### PR DESCRIPTION
**Pull request by TriplEight**
- optimize caching (swatinem's action covers more rust cache)
- cancel previous CI runs on the new commits
- pin actions' versions
- add dependabot to update cargo deps and github actions versions

Also, I'd like to use simple `run:` instead of harder-readable special actions, but it's cosmetic
I'm not quite sure if `sccache` is actually helpful here, maybe it's worth a shot next time.

### CODE REVIEW

**Comment by sorpaas** [comment]
Are you sure this works? It's cancelling its own runs.

**Comment by TriplEight** [comment]
@sorpaas fixed

**Comment by sorpaas** [comment]
There still seems to be something wrong. Any windows build always stuck at the last step.

File: .github/workflows/check-style.yml [hunk 1]
```diff
@@ -17,20 +17,25 @@ jobs:
     env:
       RUST_BACKTRACE:  full
     steps:
+      - name:          Cancel Previous Runs
+        uses:          styfle/cancel-workflow-action@0.9.1
+
       - name:          Checkout sources & submodules
-        uses:          actions/checkout@master
+        uses:          actions/checkout@v1
```
Comment (athei) [inline .github/workflows/check-style.yml:24]: I think the current checkout version is `2`. See the dependabot PR.

**Comment by sorpaas** [comment]
Maybe just remove the rust-cache step? What does it do anyway?

**Comment by athei** [comment]
Can you update `upload-artifact` to version `2`? Then we get rid of both dependabot PRs.

**Comment by TriplEight** [comment]
@athei it's already [there](https://github.com/paritytech/libsecp256k1/pull/84/commits/3034dc52e5d39df4e789e244c4e94be7a0fdbac2#diff-73e17259d77e5fbef83b2bdbbe4dc40a912f807472287f7f45b77e0cbf78792dR316)

**Comment by TriplEight** [comment]
removing the modern cache handling due to https://github.com/Swatinem/rust-cache/issues/31

**Review by sorpaas — APPROVED** [review]

**Pull request merged**